tools/expotools/src/commands/ClientBuild.ts

```diff
@@ -1,144 +1,177 @@
+    .option(
+      '-r, --release',
+      'Whether to upload and release the client build to staging versions endpoint.',
+      false
+    )
+    .option(
+      '--skip-upload',
+      "Whether to skip the uploading part. Might be useful for debugging and not to unintentionally override production's build.",
+      false
+    )
+    .asyncAction(main);
 };
 
-async function askToRecreateSimulatorBuildAsync(archivePath: string) {
+async function main(options: ActionOptions) {
+  const platform = options.platform || (await askForPlatformAsync());
+  const sdkBranchVersion = (await Git.getSDKVersionFromBranchNameAsync()) || '39.0.0';
+
+  if (options.release && !sdkBranchVersion) {
+    throw new Error(`Client builds can be released only from the release branch!`);
+  }
+
+  const builder = getBuilderForPlatform(platform);
+  const sdkVersion = sdkBranchVersion || (await askForSdkVersionAsync(platform));
+  const appVersion = await builder.getAppVersionAsync();
+
+  await buildOrUseCacheAsync(builder);
+
+  if (sdkVersion && !options.skipUpload) {
+    await uploadAsync(builder, sdkVersion, appVersion);
+  }
+  if (sdkVersion && options.release) {
+    await releaseAsync(builder, sdkVersion, appVersion);
+  }
+}
+
+function getBuilderForPlatform(platform: Platform): ClientBuilder {
+  switch (platform) {
+    case 'ios':
+      return new IosClientBuilder();
+    case 'android':
+      return new AndroidClientBuilder();
+    default: {
+      throw new Error(`Platform "${platform}" is not supported yet!`);
+    }
+  }
+}
+
+async function askToRecreateSimulatorBuildAsync(): Promise<boolean> {
   if (process.env.CI) {
     return false;
   }
   const { createNew } = await inquirer.prompt<{ createNew: boolean }>([
     {
       type: 'confirm',
       name: 'createNew',
-      message: `Simulator archive already exists at ${chalk.magenta(
-        path.relative(EXPO_DIR, archivePath)
-      )}. Do you want to create a fresh one?`,
+      message: 'Do you want to create a fresh one?',
       default: true,
     },
   ]);
   return createNew;
 }
 
-function getApplicationPathForPlatform(platform: Platform) {
-  switch (platform) {
-    case 'ios': {
-      return path.join(
-        IOS_DIR,
-        'simulator-build',
-        'Build',
-        'Products',
-        'Release-iphonesimulator',
-        'Exponent.app'
-      );
-    }
-    case 'android': {
-      return path.join(ANDROID_DIR, 'app', 'build', 'outputs', 'apk', 'release', 'app-release.apk');
-    }
-    default: {
-      throw new Error(`Platform "${platform}" is not supported yet!`);
-    }
+async function askToOverrideBuildAsync(): Promise<boolean> {
+  if (process.env.CI) {
+    return true;
   }
+  const { override } = await inquirer.prompt<{ override: boolean }>([
+    {
+      type: 'confirm',
+      name: 'override',
+      message: 'Do you want to override it?',
+      default: true,
+    },
+  ]);
+  return override;
 }
```

Comment:
Maybe it would be nice to have some kind of `factory` for prompting for confirmation? 🤔 
I imagine we could have:
```ts
function promptForConfirmationAsync(question: string, { ciDefaultValue }: { ciDefaultValue: boolean }): Promise<boolean> {
  if (process.env.CI) {
    return ciDefaultValue;
  }
  const { answer } = await inquirer.prompt<{ answer: boolean }>([
    {
      type: 'confirm',
      name: 'answer',
      message: question,
      default: true,
    },
  ]);
  return answer;
}
```

Reply:
Would you like to be a volunteer? 😄